libs/product/src/facades/composite-product/composite-product-facade.interface.ts

```diff
@@ -10,89 +10,112 @@ import { DaffCompositeProduct } from '../../models/composite-product';
 
 	/**
-	 * Get the minimum price for a composite product excluding optional items and determined by 
-	 * currently applied options.
+	 * Get the minimum discounted price for a composite product, including optional items and regardless of the current selection of item options.
+	 * This could be useful for a quick preview of the product.
 	 * @param id the id of the composite product.
 	 */
-	getMinPrice(id: string): Observable<number>;
+	getMinOptionalItemDiscountedPrice(id: string): Observable<number>;
 
 	/**
-	 * Get the maximum price for a composite product excluding optional items and determined by
-	 * currently applied options.
+	 * Get the maximum discounted price for a composite product, including optional items and regardless of the current selection of item options.
+	 * This could be useful for a quick preview of the product.
 	 * @param id the id of the composite product.
 	 */
-	getMaxPrice(id: string): Observable<number>;
+	getMaxOptionalItemDiscountedPrice(id: string): Observable<number>;
 
 	/**
-	 * Returns whether the composite product has a price range depending on applied options and excluding
-	 * optional items.
+	 * Selector for whether the composite product could have a discounted price range in any configuration,
+	 * including the optional items and regardless of the current item option selection.
+	 * This could be useful for a quick preview of the product.
 	 * @param id the id of the composite product.
 	 */
-	hasPriceRange(id: string): Observable<boolean>;
+	hasOptionalItemDiscountedPriceRange(id: string): Observable<boolean>;
 
 	/**
-	 * Get the price of a composite product based on the applied product options.
+	 * Returns whether the minimum and maximum composite product discounted prices equal the minimum and maximum
+	 * pre discount prices, including optional items. Note: this intentionally misses a usecase where there is a discount
+	 * in between the minimum and maximum prices. This is because the resulting UI would be non-sensical to the customer for
+	 * this usecase. For example, if we have a set of item options' prices: (price, discount), (10, 0), (20, 0), (15, 4).
+	 * While it is true that there is a discounted price (15, 4), the user would see the original price range (10-20) followed
+	 * by the discounted price range (10-20). This would happen, because the 15-4=11 price would fall between the two extremes
+	 * of the discounted prices and it wouldn't make sense to the customer what is happening.
 	 * @param id the id of the composite product.
 	 */
-	getPrice(id: string): Observable<number>;
+	hasOptionalItemDiscount(id: string): Observable<boolean>;
 
 	/**
-	 * Get the minimum discounted price of a composite product based on the applied product options.
+	 * Get the minimum price for a composite product, excluding unselected optional items and depending on the current selection of item options.
+	 * This would be used for a composite product that has required items without default selections.
```

Comment:
Yes it does. I meant to say that unless you have required items without default selections this value (the minimum price) will always be the same as the maximum price. Maybe there's a better way to say it. "This will be different from the 'selectorCompositeProductMaxRequiredPrice' only if there are required items without default selections."?

Reply:
I changed this a bit

Reply:
That is more clear but I think you should explicitly mention that this includes the required items' default option. Something like:
```
Get the minimum price for a composite product
This includes the current selection of item options and the default option of unselected required items but excludes unselected optional items.
This will be equal to getMaxRequiredItemPrice when all required items already have default selections.
```

Definitely more wordy but it covers the entire behavior.

Reply:
There is no such thing as a "default option of unselected required items". If an item has a default, then it has a selection by default. How about:
```
Get the minimum price for a composite product.
This treats selected item options as static prices (including default item options) and unselected required items as a range of prices.
This will be equal to `getMaxRequiredItemPrice` when all required items already have default selections.
```

Reply:
ah okay I misunderstood how it worked. I think the way you had it before was good.